pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotWindowExchangeNodeInsertRule.java

```diff
@@ -171,4 +195,59 @@ private boolean isPartitionByOnlyQuery(Window.Group windowGroup) {
     }
     return isPartitionByOnly;
   }
+
+  /**
+   * Only empty OVER() type queries can result in a situation where the LogicalProject below the LogicalWindow is an
+   * empty LogicalProject (i.e. no columns are projected). This can only occur for window functions that don't take
+   * an input column. Some example queries where this can occur are:
+   *
+   * SELECT COUNT(*) OVER() from tableName
+   * SELECT 42, COUNT(*) OVER() from tableName
+   * SELECT ROW_NUMBER() OVER() from tableName
+   *
+   * This function modifies the empty LogicalProject below the LogicalWindow to add a literal and adds a LogicalProject
+   * above LogicalWindow to remove the additional literal column from being projected any further. This also handles
+   * the addition of the LogicalExchange under the LogicalWindow.
+   *
+   * TODO: Explore an option to handle empty LogicalProject by actually projecting empty rows for each entry. This way
+   *       there will no longer be a need to add a literal to the empty LogicalProject, but just traverse the number of
+   *       rows
+   */
+  private RelNode handleEmptyProjectBelowWindow(Window window, Project project) {
+    RelOptCluster cluster = window.getCluster();
+    RexBuilder rexBuilder = cluster.getRexBuilder();
+
+    // Construct the project that goes below the window (which projects a literal)
+    final List<RexNode> expsForProjectBelowWindow = Collections.singletonList(
+        rexBuilder.makeLiteral(0, cluster.getTypeFactory().createSqlType(SqlTypeName.INTEGER)));
+    final List<String> expsFieldNamesBelowWindow = Collections.singletonList("winLiteral");
```

Comment:
Probably a dumb question - why did we settle on the approach of using hardcoded `Literal` (which I guess takes the value 0 ?) to fix this ? 

Can we not pick a random `Identifier` since the `LogicalProject` above the `LogicalWindow` is anyway going to remove it.

Reply:
I meant approach (a) in my comment. But probably fine for now. This isn't incorrect.